components/script/dom/gpucanvascontext.rs

```diff
@@ -192,110 +340,84 @@ impl GPUCanvasContextMethods for GPUCanvasContext {
-            },
-            dimension: GPUTextureDimension::_2d,
-        };
+        // Step 5
+        self.configuration.replace(Some(configuration.clone()));
 
-        // Step 8
-        let mut buffer_ids = ArrayVec::<id::BufferId, PRESENTATION_BUFFER_COUNT>::new();
-        for _ in 0..PRESENTATION_BUFFER_COUNT {
-            buffer_ids.push(self.global().wgpu_id_hub().create_buffer_id());
-        }
+        // Step 6
+        self.texture_descriptor.replace(Some(descriptor));
+
+        // Step 7
+        self.replace_drawing_buffer();
 
+        // Step 8: Validate texture descriptor
+        let texture_id = self.global().wgpu_id_hub().create_texture_id();
         self.channel
             .0
-            .send(WebGPURequest::CreateSwapChain {
+            .send(WebGPURequest::ValidateTextureDescriptor {
                 device_id: configuration.device.id().0,
-                queue_id: configuration.device.GetQueue().id().0,
-                buffer_ids,
-                context_id: self.context_id,
-                image_key: self.webrender_image,
-                format,
-                size: units::DeviceIntSize::new(size.width as i32, size.height as i32),
+                texture_id,
+                descriptor: desc,
             })
             .expect("Failed to create WebGPU SwapChain");
 
-        self.texture.set(Some(
-            &configuration.device.CreateTexture(&text_desc).unwrap(),
-        ));
-
         Ok(())
     }
 
     /// <https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-unconfigure>
     fn Unconfigure(&self) {
-        if let Some(texture) = self.texture.take() {
-            if let Err(e) = self.channel.0.send(WebGPURequest::DestroySwapChain {
-                context_id: self.context_id,
-                image_key: self.webrender_image,
-            }) {
-                warn!(
-                    "Failed to send DestroySwapChain-ImageKey({:?}) ({})",
-                    self.webrender_image, e
-                );
-            }
-            drop(texture);
-        }
+        // Step 1
+        self.configuration.take();
+        // Step 2
+        self.current_texture.take();
+        // Step 3
+        self.replace_drawing_buffer();
     }
 
     /// <https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-getcurrenttexture>
     fn GetCurrentTexture(&self) -> Fallible<DomRoot<GPUTexture>> {
-        // Step 5.
+        // Step 1
+        let configuration = self.configuration.borrow();
+        let Some(configuration) = configuration.as_ref() else {
+            return Err(Error::InvalidState);
+        };
+        // Step 2
+        let texture_descriptor = self.texture_descriptor.borrow();
+        let texture_descriptor = texture_descriptor.as_ref().unwrap();
+        // Step 6
+        let current_texture = if let Some(current_texture) = self.current_texture.get() {
+            current_texture
+        } else {
+            // Step 3&4
+            self.replace_drawing_buffer();
```

Comment:
Using this https://github.com/servo/servo/blob/c519a2cdb68683e2c9b05e54cfd10e022228fb2c/components/script/dom/resizeobserver.rs#L191 (and moving the other steps as suggested) would be more spec compliant(see the optional task queuing).